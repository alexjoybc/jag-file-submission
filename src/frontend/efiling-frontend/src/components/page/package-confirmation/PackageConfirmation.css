.inline-block {
  display: inline-block;
}

.file-href {
  cursor: pointer;
  color: #1a5a96;
  text-decoration: underline;
}

.push-margin-left {
  margin-left: 15px;
}

@media (min-width: 1200px) {
  .near-half-width {
    width: 60% !important;
  }
}
