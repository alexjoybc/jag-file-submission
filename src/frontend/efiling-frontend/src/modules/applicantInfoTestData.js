const applicantInfo = {
  bceID: "bobross42",
  firstName: "Bob",
  lastName: "Ross",
  emailAddress: "bob.ross@example.com"
};

export function getApplicantInfo() {
  return applicantInfo;
}
