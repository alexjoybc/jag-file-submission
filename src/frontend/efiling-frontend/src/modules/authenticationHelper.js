const jwt = require("jsonwebtoken");

export function getJWTData() {
  const token = localStorage.getItem("jwt");
  if (!token) return false;

  // get the decoded payload and header
  const decoded = jwt.decode(token, { complete: true });
  return decoded.payload;
}

export function generateJWTToken(payload) {
  return jwt.sign(payload, "something");
}
