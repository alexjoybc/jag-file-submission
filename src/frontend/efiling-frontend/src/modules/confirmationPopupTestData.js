/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn"
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width"
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width"
};

const handleClose = () => {};
const handleShow = () => {};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: true,
    handleShow,
    handleClose,
    handleConfirm: () => {},
    title: "title",
    body
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
