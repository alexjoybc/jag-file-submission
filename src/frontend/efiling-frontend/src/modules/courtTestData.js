const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
};

export function getCourtData() {
  return court;
}
