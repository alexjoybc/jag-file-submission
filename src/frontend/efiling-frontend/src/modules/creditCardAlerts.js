import React from "react";
import { MdCreditCard } from "react-icons/md";
import { Alert } from "shared-components";

const existingCreditCard = () => {
  return (
    <Alert
      icon={<MdCreditCard size={32} />}
      type="success"
      styling="success-background no-padding-bottom"
      element={
        <p>
          <span>
            You have a valid Credit Card registered with your CSO account.
          </span>
          <br />
          <a
            href="https://justice.gov.bc.ca/cso/about/index.do"
            target="_blank"
            rel="noopener noreferrer"
          >
            Register a new Credit Card.
          </a>
        </p>
      }
    />
  );
};

const noCreditCard = () => {
  return (
    <Alert
      icon={<MdCreditCard size={32} />}
      type="error"
      styling="error-background no-padding-bottom"
      element={
        <p>
          <span>
            You do not have a valid Credit Card registered with your CSO
            account.
          </span>
          <br />
          <a
            href="https://justice.gov.bc.ca/cso/about/index.do"
            target="_blank"
            rel="noopener noreferrer"
          >
            Register a Credit Card now
          </a>
          &nbsp;to continue.
        </p>
      }
    />
  );
};

export function getCreditCardAlerts() {
  const existingCreditCardData = existingCreditCard();
  const noCreditCardData = noCreditCard();

  return {
    existingCreditCard: existingCreditCardData,
    noCreditCard: noCreditCardData,
  };
}
