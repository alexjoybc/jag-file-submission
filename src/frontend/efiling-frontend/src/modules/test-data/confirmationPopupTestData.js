/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
