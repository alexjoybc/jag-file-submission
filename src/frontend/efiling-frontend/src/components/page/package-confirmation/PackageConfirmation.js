import React from "react";
import PropTypes from "prop-types";
import { MdInsertDriveFile } from "react-icons/md";
import ConfirmationPopup, {
  Button,
  Sidecard,
  DisplayBox
} from "shared-components";
import { getSidecardData } from "../../../modules/sidecardData";
import { propTypes } from "../../../types/propTypes";

import "./PackageConfirmation.css";

const continueButton = {
  label: "Continue to Payment",
  onClick: () => console.log("continue on click"),
  styling: "normal-blue btn"
};

const uploadButton = {
  label: "Upload Documents",
  onClick: () => console.log("upload on click"),
  styling: "normal-white btn"
};

const aboutCsoSidecard = getSidecardData().aboutCso;
const csoAccountDetailsSidecard = getSidecardData().csoAccountDetails;

const documentIcon = (
  <div style={{ color: "rgb(252, 186, 25)" }}>
    <MdInsertDriveFile size={32} />
  </div>
);

export default function PackageConfirmation({
  packageConfirmation: { confirmationPopup }
}) {
  return (
    <div className="page">
      <div className="content col-md-8">
        <h2>Package Confirmation</h2>
        <p>The following documents will be submitted to</p>
        <h3>Review your package for accuracy before submitting</h3>

        <DisplayBox
          styling="border-background"
          icon={documentIcon}
          element={<p>View</p>}
        />

        <br />

        <p>
          If you have any supporting or related documents you may upload them
          now.
        </p>

        <section className="inline-block pt-2">
          <Button
            label={uploadButton.label}
            onClick={uploadButton.onClick}
            styling={uploadButton.styling}
          />
        </section>

        <section className="buttons pt-2">
          <ConfirmationPopup
            modal={confirmationPopup.modal}
            mainButton={confirmationPopup.mainButton}
            confirmButton={confirmationPopup.confirmButton}
            cancelButton={confirmationPopup.cancelButton}
          />
          <Button
            label={continueButton.label}
            onClick={continueButton.onClick}
            styling={continueButton.styling}
          />
        </section>
      </div>

      <div className="sidecard">
        <Sidecard sideCard={csoAccountDetailsSidecard} />
        <Sidecard sideCard={aboutCsoSidecard} />
      </div>
    </div>
  );
}

PackageConfirmation.propTypes = {
  packageConfirmation: PropTypes.shape({
    confirmationPopup: propTypes.confirmationPopup
  }).isRequired
};
