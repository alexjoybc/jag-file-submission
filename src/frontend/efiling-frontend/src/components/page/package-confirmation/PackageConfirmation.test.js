import React from "react";
import { render } from "@testing-library/react";
import { getTestData } from "../../../modules/confirmationPopupTestData";

import PackageConfirmation from "./PackageConfirmation";

describe("PackageConfirmation Component", () => {
  const confirmationPopup = getTestData();
  const packageConfirmation = { confirmationPopup };
  const csoAccountStatus = { isNew: false };

  test("Matches the existing account snapshot", () => {
    const { asFragment } = render(
      <PackageConfirmation
        packageConfirmation={packageConfirmation}
        csoAccountStatus={csoAccountStatus}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Matches the new account snapshot", () => {
    const { asFragment } = render(
      <PackageConfirmation
        packageConfirmation={packageConfirmation}
        csoAccountStatus={{ ...csoAccountStatus, isNew: true }}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
