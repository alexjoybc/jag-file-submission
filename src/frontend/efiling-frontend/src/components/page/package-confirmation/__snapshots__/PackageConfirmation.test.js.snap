// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageConfirmation Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="content col-md-8"
    >
      <h2>
        Package Confirmation
      </h2>
      <p>
        The following documents will be submitted to
      </p>
      <h3>
        Review your package for accuracy before submitting
      </h3>
      <div
        class="display-box border-background"
      >
        <div
          class="display-icon"
        >
          <div
            style="color: rgb(252, 186, 25);"
          >
            <svg
              fill="currentColor"
              height="32"
              size="32"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 2c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6H6zm7 7V3.5L18.5 9H13z"
              />
            </svg>
          </div>
        </div>
        <div
          class="display-element"
        >
          <p>
            View
          </p>
        </div>
      </div>
      <br />
      <p>
        If you have any supporting or related documents you may upload them now.
      </p>
      <section
        class="inline-block pt-2"
      >
        <div>
          <button
            class="normal-white btn"
            type="button"
          >
            Upload Documents
          </button>
        </div>
      </section>
      <section
        class="buttons pt-2"
      >
        <div>
          <button
            class="normal-white btn"
            type="button"
          >
            Cancel
          </button>
        </div>
        <div>
          <button
            class="normal-blue btn"
            type="button"
          >
            Continue to Payment
          </button>
        </div>
      </section>
    </div>
    <div
      class="sidecard"
    >
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        Your CSO Account
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    CSO account 
                    <b>
                      blah
                    </b>
                     is linked to your Basic BCeID account 
                    <b>
                      blah2
                    </b>
                     and will be used to file documents. 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      View your CSO account details
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
      <div
        class="wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="row"
        >
          <div
            class="col-12 pt-2"
            style="position: relative;"
          >
            <p />
            <section
              class="bluegrey-container"
              id="bluegrey-section"
            >
              <div
                class="container-background bluegrey-heading"
              >
                <h2
                  class="heading-style"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="side-card-row"
                    >
                      <div
                        class="round-icon-wrapper"
                      >
                        <svg
                          class="side-card-icon"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
                          />
                        </svg>
                      </div>
                      <div
                        class="side-card-title"
                      >
                        About E-File Submission
                      </div>
                    </div>
                  </div>
                </h2>
              </div>
              <div
                class="bluegrey-content"
              >
                <span
                  class="content-style"
                >
                  <p>
                    E-File submission is a service to help you securely and electronically file documents with the Government of British Columbia Court Services Online (CSO). 
                    <a
                      href="https://justice.gov.bc.ca/cso/about/index.do"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn more about CSO
                    </a>
                    .
                  </p>
                </span>
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
